src/python/WMCore/Services/WMArchive/DataMap.py

```diff
@@ -192,12 +194,14 @@ def typeCastPerformance(performDict):
                         value = performDict[key][param]
                         if value in ["-nan", "nan", "inf", ""]:
                             value = -1
-                        performDict[key][param] = PERFORMANCE_TYPE[key][param](value)
+                        if PERFORMANCE_TYPE[key][param] == int:
```

Comment:
that's fine, but I would suggest to add explicit comment here to clarify what code is doing. Something like:
```
# the received value comes from CMSSW FWJR and its type is string
# first we convert it to float because int(string) may lead to error, e.g. int("3.0") will not work
# then we call PERFORMANCE_TYPE[key][param] converter over the value, in this case
# PERFORMANCE_TYPE[key][param] is int. The full chain would be int(float(string))
```